src/algebra/algebra/spectrum.lean

```diff
@@ -0,0 +1,195 @@
+/-
+Copyright (c) 2021 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import algebra.algebra.basic
+import tactic.noncomm_ring
+/-!
+# Spectrum of an element in an algebra
+This file develops the basic theory of the spectrum of an element of an algebra.
+This theory will serve as the foundation for spectral theory in Banach algebras.
+
+## Main definitions
+
+* `resolvent a`: has type `set R` is the resolvent set of an element `a : A` where
+  `A` is an  `R`-algebra.
+* `spectrum a`: has type `set R` is the spectrum of an element `a : A` where
+  `A` is an  `R`-algebra.
+
+## Main statements
+
+* `unit_left_coset_spectrum`: units in the scalar ring commute (multplication) with the spectrum.
+* `left_add_coset_spectrum`: elements of the scalar ring commute (addition) with the spectrum.
+* `unit_mem_spectrum_mul_iff_swap_mul`: the units (of `R`) in `σ (a*b)` coincide
+  with those in `σ (b*a)`.
+
+## Notations
+
+* `σ a` : `spectrum R a` of `a : A`
+
+## TODO
+
+* Prove the *spectral mapping theorem* for the polynomial functional calculus
+-/
+
+universes u v
+
+namespace spectrum
```

Comment:
I've thought a bit more about the namespace issue, and it looks like the main "competitor" is already called `prime_spectrum`, so it's probably not too crazy to put `spectrum` directly in the root namespace. It looks like `resolvent` should be OK too -- perhaps we could call it `resolvent_set` to make sure. Maybe we could ask on Zulip and see if anyone complains.

Given this though, what goes in the `spectrum` namespace is still on a case-by-case basis, I don't think it makes sense to put everything in this file under that namespace, especially if it leads to duplication of names (i.e. we could have either `mem_spectrum_iff` or `spectrum.mem_iff`, but `spectrum.mem_spectrum_iff` seems unnecessarily long). Also, the lemmas pertaining to `is_unit` shouldn't be under `spectrum` either (see my other comments).

Reply:
I think I have addressed all the name/namespace issues now. Declarations in the `spectrum` namespace no longer include that word in their names, `spectrum` and `resolvent` both appear in the root namespace, and the two `is_unit` lemmas are no longer in the `spectrum` namespace.